### PR TITLE
DOT-4816: Webhook to check the status of orders processed by Cash Plus

### PR DESCRIPTION
**Pull request by olyans**

### CODE REVIEW

File: includes/admin/youcanpay-settings.php [hunk 1]
```diff
@@ -62,6 +62,11 @@
 			'default'     => '',
 			'desc_tip'    => true,
 		],
+		'webhook'                      => [
```
Comment (zahraoui-mohammed) [inline includes/admin/youcanpay-settings.php:65]: Apply coding style

File: includes/class-wc-youcanpay-webhook-handler.php [hunk 2]
```diff
@@ -38,30 +38,148 @@ public function __construct() {
...
+
+            if (!array_key_exists('transaction', $data['payload'])) {
+                return false;
+            }
+
+            if (!array_key_exists('payment_method', $data['payload'])) {
+                return false;
+            }
+
+            $transaction_id = null;
```
Comment (zahraoui-mohammed) [inline includes/class-wc-youcanpay-webhook-handler.php:99]: Apply coding style by using camel case

File: includes/class-wc-youcanpay-webhook-handler.php [hunk 3]
```diff
@@ -38,30 +38,148 @@ public function __construct() {
...
+            $transaction = new WC_YouCanPay_Transaction_Model($data['payload']['transaction']);
+            $payment_method = new WC_YouCanPay_Payment_Method_Model($data['payload']['payment_method']);
+            $payment_method_name = sprintf(__('YouCan Pay (%s)', 'youcan-pay'), $payment_method->get_name());
```
Comment (zahraoui-mohammed) [inline includes/class-wc-youcanpay-webhook-handler.php:102]: Apply coding style by using camel case

File: includes/class-wc-youcanpay-webhook-handler.php [hunk 4]
```diff
@@ -38,30 +38,148 @@ public function __construct() {
...
+                return false;
+            }
+
+            if (!array_key_exists('payment_method', $data['payload'])) {
+                return false;
+            }
+
+            $transaction_id = null;
+            $transaction = new WC_YouCanPay_Transaction_Model($data['payload']['transaction']);
+            $payment_method = new WC_YouCanPay_Payment_Method_Model($data['payload']['payment_method']);
```
Comment (zahraoui-mohammed) [inline includes/class-wc-youcanpay-webhook-handler.php:101]: Apply coding style by using camel case

File: includes/class-wc-youcanpay-webhook-handler.php [hunk 5]
```diff
@@ -38,30 +38,148 @@ public function __construct() {
...
+                WC_YouCanPay_Logger::info('payment successfully processed', [
+                    'payment_method' => $payment_method_name,
+                    'transaction_id' => $transaction->get_id(),
+                    'order_id'       => $order->get_id(),
+                    'order_total'    => $order->get_total(),
+                ]);
+
+                //TODO: Need to using WC_YouCanPay_Helper::set_payment_method_to_order() instead of $order->set_payment_method()
+
+                $order->set_payment_method($payment_method_name);
```
Comment (zahraoui-mohammed) [inline includes/class-wc-youcanpay-webhook-handler.php:150]: Apply coding style by using camel case for naming this method

File: includes/class-wc-youcanpay-webhook-handler.php [hunk 6]
```diff
@@ -38,30 +38,148 @@ public function __construct() {
...
+                //WC_YouCanPay_Helper::set_payment_method_to_order( $order, WC_Gateway_YouCanPay::ID );
+                $order->payment_complete($transaction->get_id());
```
Comment (zahraoui-mohammed) [inline includes/class-wc-youcanpay-webhook-handler.php:152]: Apply coding style by using camel case for naming this method.

Comment (zahraoui-mohammed) [reply includes/class-wc-youcanpay-webhook-handler.php:152]: Also, use an action verb when naming methods.  Like completePayment()

**Review by zahraoui-mohammed — CHANGES_REQUESTED** [review]
Left some comments

File: includes/class-wc-youcanpay-webhook-handler.php (hunk 3 above)
Comment (olyans) [reply includes/class-wc-youcanpay-webhook-handler.php:102]: PHP Coding Standards for WordPress

Use lowercase letters in variable, action/filter, and function names (never camelCase). Separate words via underscores. Don’t abbreviate variable names unnecessarily; let the code be unambiguous and self-documenting.

File: includes/class-wc-youcanpay-webhook-handler.php (hunk 4 above)
Comment (olyans) [reply includes/class-wc-youcanpay-webhook-handler.php:101]: PHP Coding Standards for WordPress

Use lowercase letters in variable, action/filter, and function names (never camelCase). Separate words via underscores. Don’t abbreviate variable names unnecessarily; let the code be unambiguous and self-documenting.

File: includes/class-wc-youcanpay-webhook-handler.php (hunk 5 above)
Comment (olyans) [reply includes/class-wc-youcanpay-webhook-handler.php:150]: PHP Coding Standards for WordPress

Use lowercase letters in variable, action/filter, and function names (never camelCase). Separate words via underscores. Don’t abbreviate variable names unnecessarily; let the code be unambiguous and self-documenting.

File: includes/class-wc-youcanpay-webhook-handler.php (hunk 6 above)
Comment (olyans) [reply includes/class-wc-youcanpay-webhook-handler.php:152]: is a native method of woocommerce

File: includes/class-wc-youcanpay-webhook-handler.php (hunk 2 above)
Comment (olyans) [reply includes/class-wc-youcanpay-webhook-handler.php:99]: PHP Coding Standards for WordPress

Use lowercase letters in variable, action/filter, and function names (never camelCase). Separate words via underscores. Don’t abbreviate variable names unnecessarily; let the code be unambiguous and self-documenting.

File: includes/admin/youcanpay-settings.php (hunk 1 above)
Comment (olyans) [reply includes/admin/youcanpay-settings.php]: https://github.com/NextmediaMa/youcan-pay-for-woocommerce/pull/16/commits/56a48fbf26951c53184bc6167976627a992d5a01

**Review by zahraoui-mohammed — APPROVED** [review]

**Pull request merged**